Dashboard/app/js/lib/views/views.js

```diff
@@ -87,6 +87,47 @@ Ember.Handlebars.registerHelper('tooltip', function (i18nKey) {
 });
 
 
+FLOW.renderCaddisflyAnswer = function(json){
+  var name = ""
+  var imageUrl = ""
+  var result = Ember.A();
+  if (!Ember.empty(json)){
+    jsonParsed = JSON.parse(json);
+
+    // get out test name
+    if (!Ember.empty(jsonParsed.name)){
+      testName =  jsonParsed.name.trim();
+    }
+
+    // get out results
+    if (jsonParsed.result && !Ember.empty(jsonParsed.result)){
+        jsonParsed.result.forEach(function(item){
+        newResult = {
+          "name":item.name,
+          "value":item.value,
+          "unit":item.unit,
+        };
+        result.push(newResult);
+      });
+    }
+    // get out image url
+    if (!Ember.empty(jsonParsed.image)){
+      imageUrl = FLOW.Env.photo_url_root + jsonParsed.image.trim();
+    }
+
+    // contruct html
+    html = "<div><strong>" + name + "</strong></div>"
+    html += result.map(function(item){
```

Comment:
It seems that there is no need to have an intermediate copy of the `jsonParsed.result` in `result` if that is already an array and can be used with `map`?